add cron to docker compose file

segunfUWaterloo opened the pull request:
## Summary

- Adds the analytics cron 
- Adds the endpoint to fetch the analytics data per user

## Tests

- Test for fetching an invalid user
- Test for fetching a valid user
- Test for weekly and daily reset with the cron
- Test for the update of the analytics table when a new session is completed 
- Test the cli parameters for the cron

segunfUWaterloo committed:
add cron to docker compose file

LokiWager commented on cron/db/__init__.py:
```diff
@@ -0,0 +1,53 @@
+                    connectTimeoutMS=3000,
+                )
+            cls._instance.db = cls._instance.client[cls._instance.cfg.db]
+            cls._instance._init_index(
+                "blocklist",
+                [
+                    ("user_id", ASCENDING),
+                    ("domain", ASCENDING),
+                    ("list_type", ASCENDING),
+                ],
+            )
```
You need to add the index that you want.

segunfUWaterloo committed:
Adding the completed_sessions field to list analytics
Updating the tests for the new field

segunfUWaterloo committed:
updating the analytics svc response

segunfUWaterloo committed:
Update the db init index to initialized analytics index
Set the update fuction to update the completed sessions field
Update the test to ensure completed sessions is incremented

segunfUWaterloo committed:
updating requirement.txt

segunfUWaterloo committed:
Updating the tests to use instance methods
Adding the indexes to the db init
Adding logging

segunfUWaterloo committed:
sync changes from main

segunfUWaterloo committed:
setting default log level to warn

segunfUWaterloo committed:
Add try/except for db inserts/update
Update tests to current serially

segunfUWaterloo committed:
remove unused variables

LokiWager reviewed (approved):
lgtm
